src/models/backbones/resnet.py

```diff
@@ -0,0 +1,342 @@
+"""TorchOK ResNet
+
+Adapted from https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/resnet.py
+Copyright 2019 Ross Wightman
+Licensed under The Apache 2.0 License [see LICENSE for details]
+"""
+from typing import Optional, Union, List, Dict
+
+import torch
+import torch.nn as nn
+
+from src.constructor import BACKBONES
+from src.models.modules.blocks.se import SEModule
+from src.models.modules.bricks.convbnact import ConvBnAct
+from src.models.base_model import BaseModel, FeatureInfo
+from src.models.backbones.utils.helpers import build_model_with_cfg
+from src.models.backbones.utils.constants import IMAGENET_DEFAULT_STD, IMAGENET_DEFAULT_MEAN
+
+
+def _cfg(url='', **kwargs):
+    return {
+        'url': url,
+        'input_size': (3, 224, 224),
+        'pool_size': (7, 7),
+        'crop_pct': 0.875,
+        'interpolation': 'bilinear',
+        'mean': IMAGENET_DEFAULT_MEAN,
+        'std': IMAGENET_DEFAULT_STD,
+        'first_conv': 'conv1',
+        **kwargs
+    }
+
+
+default_cfgs = {
+    # ResNet
+    'resnet18': _cfg(url='https://torchok-hub.s3.eu-west-1.amazonaws.com/resnet18-torchok.pth'),
+    'resnet34': _cfg(
+        url=''),
+    'resnet50': _cfg(
+        url='https://torchok-hub.s3.eu-west-1.amazonaws.com/resnet50-torchok.pth',
+        interpolation='bicubic',
+        crop_pct=0.95),
+    'resnet101': _cfg(
+        url='',
+        interpolation='bicubic',
+        crop_pct=0.95),
+    'resnet152': _cfg(
+        url='',
+        interpolation='bicubic',
+        crop_pct=0.95),
+
+    # Squeeze-Excitation ResNets
+    'seresnet18': _cfg(interpolation='bicubic'),
+    'seresnet34': _cfg(interpolation='bicubic'),
+    'seresnet50': _cfg(
+        url='',
+        interpolation='bicubic'),
+    'seresnet101': _cfg(interpolation='bicubic'),
+    'seresnet152': _cfg(interpolation='bicubic')
+}
+
+
+class BasicBlock(nn.Module):
+    """BasicBlock bulding block for ResNet architecture."""
+
+    expansion = 1
+
+    def __init__(self,
+                 in_channels: int,
+                 out_channels: int,
+                 stride: int = 1,
+                 downsample: Optional[nn.Module] = None,
+                 attn_layer: Optional[nn.Module] = None):
+        """Init BasicBlock.
+
+        Args:
+            in_channels: Number of input channels.
+            out_channels: Number of output channels.
+            stride: Stride.
+            downsample: Downsample module.
+            attn_layer: Attention block.
+        """
+        super().__init__()
+        out_block_channels = out_channels * self.expansion
+
+        self.convbnact = ConvBnAct(in_channels, out_channels, kernel_size=3, padding=1, stride=stride)
+        self.conv = nn.Conv2d(out_channels, out_block_channels, kernel_size=3, padding=1, bias=False)
```

Comment:
How you look at idea to add functional in ConvBnAct to create briks without activation layer. So you can add ConvBnAct for conv and bn too.